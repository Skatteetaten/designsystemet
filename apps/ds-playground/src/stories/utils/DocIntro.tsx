import { ReactNode } from 'react';

import { Link } from '@skatteetaten/ds-buttons';
import { Paragraph } from '@skatteetaten/ds-typography';

interface DocIntroProps {
  name: string;
  url: string;
}

export const DocIntro = ({ name, url }: DocIntroProps): ReactNode => {
  return (
    <Paragraph variant={'ingress'}>
      {`Se `}
      <Link href={url}>{name}</Link>
      {` p√• stil og tone for eksempler og
        retningslinjer for denne komponenten.`}
    </Paragraph>
  );
};
