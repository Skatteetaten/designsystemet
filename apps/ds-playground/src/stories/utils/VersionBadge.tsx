import { ReactNode } from 'react';

interface VersionBadgeProps {
  packageName: string;
}

export const VersionBadge = ({ packageName }: VersionBadgeProps): ReactNode => {
  let color = 'green';

  let version =
    import.meta.env.DS_VERSIONS[`${packageName.replaceAll('-', '_')}`] ?? '..';
  if (import.meta.env.MODE === 'development') {
    color = 'red';
    version = 'developing';
  }
  if (import.meta.env['STORYBOOK_BUILD_VERSION']?.includes('SNAPSHOT')) {
    color = 'blue';
  }

  const normalizedPackageName = packageName.replaceAll('-', '--');
  const normalizedVersion = version.replaceAll('-', '--').replaceAll('_', '__');

  return (
    <img
      src={`https://img.shields.io/badge/%40skatteetaten%2Fds--${normalizedPackageName}-${normalizedVersion}-${color}?logo=npm`}
      alt={`@skatteetaten/ds-${packageName}@${version}`}
    />
  );
};
