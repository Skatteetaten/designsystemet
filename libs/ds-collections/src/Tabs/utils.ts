export const valueRegex = /^[A-Za-z][\w\-.:]*$/;
