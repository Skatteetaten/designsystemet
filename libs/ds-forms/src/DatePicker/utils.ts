import { format, isValid, parse } from 'date-fns';

export const formatDateForInput = (
  dateFormat: string,
  date?: Date | null
): string => {
  return !isValid(date) ? '' : format(date as Date, dateFormat);
};

const COMMON_INPUT_FORMATS = [
  'dd.MM.yy',
  'dd/MM/yy',
  'dd-MM-yy',
  'dd.MM.yyyy',
  'dd/MM/yyyy',
  'dd-MM-yyyy',
  'ddMM',
  'ddMMyy',
  'ddMMyyyy',
];

export const parseDateFromInput = (
  value: string,
  dateFormat: string
): Date | null => {
  const dateFormats = Array.from(
    new Set([...COMMON_INPUT_FORMATS, dateFormat])
  );

  for (const format of dateFormats) {
    const date = parse(value, format, new Date());
    if (isValid(date)) {
      return date;
    }
  }

  return null;
};
