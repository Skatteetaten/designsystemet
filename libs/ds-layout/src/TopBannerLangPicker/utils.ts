import { Languages } from '@skatteetaten/ds-core-utils';

import { TopBannerLangPickerProps } from './TopBannerLangPicker.types';

export const convertLocaleToLang = (locale: Languages): string => {
  return locale.substring(0, locale.indexOf('_'));
};

export type LanguageItem = {
  lang: string;
  displayName: string;
};

export type LanguageItems = {
  [key: string]: LanguageItem;
};

export const isLanguages = (value: unknown): value is Languages => {
  return Object.values(Languages).includes(value as Languages);
};

export const getCurrentLanguages = (
  includeSami: boolean,
  additionalLanguages: TopBannerLangPickerProps['additionalLanguages'] = []
): LanguageItems => {
  const defaultLanguages: LanguageItems = {
    nb: {
      lang: 'nb',
      displayName: 'Bokmål',
    },
    nn: {
      lang: 'nn',
      displayName: 'Nynorsk',
    },
    en: {
      lang: 'en',
      displayName: 'English',
    },
  };
  const se = {
    se: {
      lang: 'se',
      displayName: 'Sámegiella',
    },
  };

  const additionalLanguagesAsObject =
    additionalLanguages?.reduce<LanguageItems>((acc, curr) => {
      acc[curr.lang] = curr;
      return acc;
    }, {});

  return includeSami
    ? { ...defaultLanguages, ...se, ...additionalLanguagesAsObject }
    : { ...defaultLanguages, ...additionalLanguagesAsObject };
};
