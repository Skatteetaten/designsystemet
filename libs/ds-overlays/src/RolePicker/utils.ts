import { formatter } from '@skatteetaten/ds-core-utils';

export const formatOrganisationNumber = (
  organisationNumber: string
): string => {
  const result = formatter({
    type: 'organisationNumber',
    value: organisationNumber,
  });

  return result.value;
};

export const formatNationalIdentityNumber = (
  nationalIdentityNumber: string
): string => {
  const result = formatter({
    type: 'nationalIdentityNumber',
    value: nationalIdentityNumber,
  });

  return result.value;
};
