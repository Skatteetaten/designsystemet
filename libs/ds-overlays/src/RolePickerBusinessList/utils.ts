import { dsI18n } from '@skatteetaten/ds-core-utils';

import { Business } from '../RolePicker/RolePicker.types';

const unitTypesToHide = [
  'AS',
  'DA',
  'ANS',
  'ASA',
  'SE',
  'BA',
  'SA',
  'SCE',
  'STI',
  'IKS',
  'KF',
  'FKF',
  'RHF',
  'HF',
  'NUF',
];

export const getBusinessTitle = (business: Business): string => {
  let title = business.name;
  if (!unitTypesToHide.includes(business.unitType)) {
    title += ` ${business.unitType}`;
  }

  if (business.isDeleted) {
    title += ` (${dsI18n.t('ds_overlays:rolepicker.Deleted')})`;
  }

  return title;
};
