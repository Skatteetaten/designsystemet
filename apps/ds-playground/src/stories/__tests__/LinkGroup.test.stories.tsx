import { Meta, StoryFn, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import {
  LinkGroup,
  LinkGroupProps,
  linkGroupVariantArr,
} from '@skatteetaten/ds-buttons';
import { linkColorArr } from '@skatteetaten/ds-core-utils';
import { CalendarSVGpath } from '@skatteetaten/ds-icons';

import { wrapper } from './testUtils/storybook.testing.utils';

const meta = {
  component: LinkGroup,
  title: 'Tester/LinkGroup',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: { disable: true },
    },
    hasSpacing: { table: { disable: true } },
    color: {
      table: { disable: true },
      options: [undefined, ...linkColorArr],
      control: 'inline-radio',
    },
    variant: {
      table: { disable: true },
      options: [...linkGroupVariantArr],
      control: 'inline-radio',
    },
  },
} satisfies Meta<typeof LinkGroup>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs: LinkGroupProps = {
  children: [
    <LinkGroup.Link
      key={'linkGroupLink_1'}
      href={'#storybook-root'}
      onClick={(e): void => e.preventDefault()}
    >
      {'Er du pendler?'}
    </LinkGroup.Link>,
    <LinkGroup.Link
      key={'linkGroupLink_2'}
      href={'#storybook-root'}
      onClick={(e): void => e.preventDefault()}
    >
      {'Pendler du mye?'}
    </LinkGroup.Link>,
    <LinkGroup.Link
      key={'linkGroupLink_3'}
      href={'#storybook-root'}
      onClick={(e): void => e.preventDefault()}
    >
      {'Pendler du dagen lang?'}
    </LinkGroup.Link>,
  ],
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLUListElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const linkGroup = canvas.getByRole('list');
    await expect(linkGroup).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const linkGroup = canvas.getByRole('list');
    await expect(linkGroup).toHaveClass('dummyClassname');
    await expect(linkGroup).toHaveAttribute('id', 'htmlId');
    await expect(linkGroup).toHaveAttribute('lang', 'nb');
    await expect(linkGroup).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults Variant List (A1 delvis, A4 delvis, A5)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    children: {
      table: { disable: false },
    },
  },
  parameters: {
    imageSnapshot: {
      hover: `${wrapper} > ul > li:first-child > a`,
      focus: `${wrapper} > ul > li:first-child > a`,
      click: `${wrapper} > ul > li:first-child > a`,
    },
  },
} satisfies Story;

export const VariantAnchors = {
  name: 'Variant Anchors (A1 delvis, A4 delvis, A6)',
  args: {
    ...defaultArgs,
    variant: 'anchors',
  },
  argTypes: {
    variant: {
      table: {
        disable: false,
      },
    },
  },
  parameters: {
    imageSnapshot: {
      hover: `${wrapper} > ul > li:first-child > a`,
    },
  },
} satisfies Story;

export const WithLongTextAndIcons = {
  name: 'With Long Text And Icons (A2)',
  args: {
    ...defaultArgs,
    children: [
      <LinkGroup.Link
        key={'linkGroupLink_1'}
        href={'#storybook-root'}
        isExternal
      >
        {'Denne lenken har en veldig lang tekst. S책 lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert ' +
          'uansett om ikon eller ikke og f책r et hengende innrykk.'}
      </LinkGroup.Link>,
      <LinkGroup.Link
        key={'linkGroupLink_2'}
        href={'#storybook-root'}
        svgPath={CalendarSVGpath}
      >
        {'Denne lenken har en veldig lang tekst. S책 lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert ' +
          'uansett om ikon eller ikke og f책r et hengende innrykk.'}
      </LinkGroup.Link>,
    ],
  },
  argTypes: {
    children: { table: { disable: false } },
  },
} satisfies Story;

const TemplateWithTwoLinkGroups: StoryFn<typeof LinkGroup> = (args) => (
  <>
    <LinkGroup {...args} />
    <LinkGroup {...args} />
  </>
);

export const WithSpacing = {
  render: TemplateWithTwoLinkGroups,
  name: 'With Spacing (A3)',
  args: {
    ...defaultArgs,
    hasSpacing: true,
  },
  argTypes: {
    hasSpacing: { table: { disable: false } },
  },
} satisfies Story;

export const WithColor = {
  name: 'With Color (A7)',
  args: {
    ...defaultArgs,
    color: 'white',
  },
  argTypes: {
    color: {
      table: { disable: false },
    },
  },
  parameters: {
    backgrounds: {
      default: 'themePrimary',
    },
    imageSnapshot: {
      hover: `${wrapper} > ul > li:first-child > a`,
      focus: `${wrapper} > ul > li:first-child > a`,
      click: `${wrapper} > ul > li:first-child > a`,
    },
  },
} satisfies Story;
