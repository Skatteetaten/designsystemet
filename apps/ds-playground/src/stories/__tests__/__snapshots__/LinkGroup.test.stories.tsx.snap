// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/LinkGroup Defaults smoke-test 1`] = `
<div data-test-block="true">
  <ul class="_linkGroup_w23nb_1   ">
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup VariantAnchors smoke-test 1`] = `
<div data-test-block="true">
  <ul class="_linkGroup_w23nb_1   ">
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_down_w23nb_29 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M4 14.64L12 21.5L20 14.64L18.33 12.7L13.25 17.05L13.25 3.5H10.75L10.75 17.05L5.67 12.7L4 14.64Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_down_w23nb_29 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M4 14.64L12 21.5L20 14.64L18.33 12.7L13.25 17.05L13.25 3.5H10.75L10.75 17.05L5.67 12.7L4 14.64Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_down_w23nb_29 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M4 14.64L12 21.5L20 14.64L18.33 12.7L13.25 17.05L13.25 3.5H10.75L10.75 17.05L5.67 12.7L4 14.64Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithAttributes play-test 1`] = `
<div data-test-block="true">
  <ul id="htmlId"
      class="_linkGroup_w23nb_1  dummyClassname "
      lang="nb"
      data-testid="123ID"
  >
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithColor smoke-test 1`] = `
<div data-test-block="true">
  <ul class="_linkGroup_w23nb_1   ">
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 _icon_white_w23nb_23"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1 _link_white_z7jba_1 "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 _icon_white_w23nb_23"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1 _link_white_z7jba_1 "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 _icon_white_w23nb_23"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1 _link_white_z7jba_1 "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithLongTextAndIcons smoke-test 1`] = `
<div data-test-block="true">
  <ul class="_linkGroup_w23nb_1   ">
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Denne lenken har en veldig lang tekst. S책 lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert uansett om ikon eller ikke og f책r et hengende innrykk.
        <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_z7jba_35 _icon_isExternal_z7jba_44"
             viewbox="0 0 24 24"
             focusable="false"
             aria-label="Til et annet nettsted"
             aria-hidden="false"
             role="img"
        >
          <path d="M14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7Zm5 16H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7h-2v7Z">
          </path>
        </svg>
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_z7jba_35 _icon_isCustom_z7jba_41"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M19 19H5V8h14v11ZM16 1v2H8V1H6v2H5c-1.11 0-2 .89-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2h-1V1h-2Zm1 11h-5v5h5v-5Z">
          </path>
        </svg>
        Denne lenken har en veldig lang tekst. S책 lang at den lange teksten tvinger fram linjeskift med tekst som alltid er venstrejustert uansett om ikon eller ikke og f책r et hengende innrykk.
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithRef play-test 1`] = `
<div data-test-block="true">
  <ul class="_linkGroup_w23nb_1   "
      id="dummyIdForwardedFromRef"
  >
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;

exports[`Tester/LinkGroup WithSpacing smoke-test 1`] = `
<div data-test-block="true">
  <ul class="_linkGroup_w23nb_1 _linkGroup_hasSpacing_w23nb_5  ">
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
  <ul class="_linkGroup_w23nb_1 _linkGroup_hasSpacing_w23nb_5  ">
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Er du pendler?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du mye?
      </a>
    </li>
    <li class="_linkGroupItem_w23nb_8">
      <svg class="_systemIcon_6zbcy_2 _systemIcon_medium_6zbcy_9 _icon_w23nb_14 _icon_forward_w23nb_26 "
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M14.14 20.5L21 12.5L14.14 4.5L12.2 6.17L16.55 11.25L3 11.25V13.75L16.55 13.75L12.2 18.83L14.14 20.5Z">
        </path>
      </svg>
      <a class="_link_z7jba_1  "
         href="#storybook-root"
      >
        Pendler du dagen lang?
      </a>
    </li>
  </ul>
</div>
`;
